diff --git a/src/i2pd/Makefile.linux b/src/i2pd/Makefile.linux
index 4ea39e22..3469310b 100644
--- a/src/i2pd/Makefile.linux
+++ b/src/i2pd/Makefile.linux
@@ -40,7 +40,7 @@ ifeq ($(USE_UPNP),yes)
 endif
 	LDLIBS += -lpthread -ldl
 else
-	LDLIBS += -lssl -lcrypto -lz -lboost_program_options -lpthread -latomic
+	LDLIBS += -L/app/lib -lssl -lcrypto -lz -lboost_program_options -lpthread -latomic
 ifeq ($(USE_UPNP),yes)
 	LDLIBS += -lminiupnpc
 endif
